### ISSUE TITLE
Can I capture render result?

### ISSUE THREAD

**Issue by zimond** [Nazariglez/notan#106]
Can I create a CPU buffer and render to it, So I can save the render result to an image? I'm not familiar with notan, and sorry if this is noob question.

**Comment by Nazariglez** [comment]
Yes! You can use the `image` crate to export the content of a texture to the filedisk. Check this example:

```
use notan::draw::*;
use notan::prelude::*;

#[notan_main]
fn main() -> Result<(), String> {
    notan::init_with(setup)
        .add_config(DrawConfig) // Simple way to add the draw extension
        .build()
}

fn setup(app: &mut App, gfx: &mut Graphics) {
    // create a render texture using the screen's size
    let (width, height) = gfx.size();
    let mut rt = gfx.create_render_texture(width, height).build().unwrap();

    // draw a triangle
    let mut draw = gfx.create_draw();
    draw.clear(Color::BLACK);
    draw.triangle((400.0, 100.0), (100.0, 500.0), (700.0, 500.0))
        .color_vertex(Color::RED, Color::BLUE, Color::GREEN);

    // render the triangle into the RenderTexture
    gfx.render_to(&mut rt, &draw);

    // read the pixels from the RenderTexture
    let mut pixels = vec![0; (width * height * 4) as usize];
    gfx.read_pixels(&rt).read_to(&mut pixels).unwrap();

    // use the image crate to export the buffer to the filedisk
    image::save_buffer("triangle.png", &pixels, width as _, height as _, image::ColorType::Rgba8).unwrap();

    // close the app
    app.exit();
}
```

This will create a png with this: 
![triangle](https://user-images.githubusercontent.com/1929191/175045949-bab34031-5688-4c24-8665-193ce0792c10.png)

This solution will not work on the web, if you need to do this on the web let me know, maybe we can create some custom solution integrated in notan to do it. 

Feel free to close this is it resolve your question, have a nice day!

**Comment by Nazariglez** [comment]
Just to be sure, I may be wrong with your question, but I am talking about render to GPU and then move it to the CPU and exporting it to a file. Is that your use case? If you don't have a GPU notan will not work for you because all the rendering happens in the GPU. 

Anyway, going back to the question.

The problem is not the image crate but the filesystem and how the browsers work. Save a file on browsers is "hard" and saving an image can be tricky because of encoding. 

So, there are two options:
1. We can render the textures to an offscreen canvas and then use the canvas to create an `url object` and force the download. The heavy lifting of encoding will be done by the browser here, but we need at least a secondary (hidden) canvas to do it. 
2. We can use `image` crate, do the encoding with it, and then create a js `Blob` object, the blob to an `url object` and force the download. This seems cleaner (though maybe `image` crate is a little slower than the encoding systems of the browser itself.)

This can be a nice feature behind a feature flag. What do you think? If this is useful for you I can try to do it for the develop branch with a simple API that works across platforms.

**Comment by zimond** [comment]
I see why you are feeling confused. I'm actually trying to export image buffer for other jobs like image analyzing and more.  I think this is quite straightforward (after I know how to do it LOL) so I won't bother you to add a new API.  Let's just close this issue and thanks for your explanation!

**Comment by Nazariglez** [comment]
Hey @zimond I thought that this was a cool feature and I added it behind the `texture_to_file` feature in the develop branch. 

The same example now will be:
```
use notan::draw::*;
use notan::prelude::*;

#[notan_main]
fn main() -> Result<(), String> {
    notan::init_with(setup)
        .add_config(DrawConfig) // Simple way to add the draw extension
        .build()
}

fn setup(app: &mut App, gfx: &mut Graphics) {
    // create a render texture using the screen's size
    let (width, height) = gfx.size();
    let mut rt = gfx.create_render_texture(width, height).build().unwrap();

    // draw a triangle
    let mut draw = gfx.create_draw();
    draw.clear(Color::BLACK);
    draw.triangle((400.0, 100.0), (100.0, 500.0), (700.0, 500.0))
        .color_vertex(Color::RED, Color::BLUE, Color::GREEN);

    // render the triangle into the RenderTexture
    gfx.render_to(&mut rt, &draw);

    // export texture content to a file 
    rt.to_file(gfx, "triangle.png").unwrap();

    // close the app
    app.exit();
}
```

It will take care of flipping the output for the RenderTexture and It works on browsers and desktop. I hope it helps.

**Comment by zimond** [comment]
Thanks!
